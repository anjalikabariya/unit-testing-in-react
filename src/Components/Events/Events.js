import React from 'react'
import { Button } from '@mui/material';


export default function Events() {

    const [counter, setcounter] = React.useState(0);
    return (
        <div className='Counter'>
            <h1 data-test='counter-title'>Event & Snapshot Test</h1>
            
            <h2 data-test='counter-value'>{counter}</h2>

            <Button variant='contained' color='primary' data-test='increment-btn' onClick={() => setcounter(counter + 1)} >Increment</Button>&nbsp;
            <Button variant='contained' color='primary' data-test='decrement-btn' onClick={() => setcounter((counter > 0) ? (counter - 1) : 0)} >Decrement</Button>
        </div>
    )
}
