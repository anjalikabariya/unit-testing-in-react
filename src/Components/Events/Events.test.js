import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Events from './Events'

/**
 * Snapshot tests are a useful when UI does not change frequently.
 *
 * A typical snapshot test case for a mobile app renders a UI component, takes a snapshot, 
 * then compares it to a reference snapshot file stored alongside the test.
 */
describe('Events Component', () => {

    let wrapper
    beforeEach(() => {
        wrapper = shallow(<Events />)
    })

    test('Matches the snapshot', () => {
      const tree = renderer.create(<Events />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    test('title', () => {
      let title = wrapper.find(`[data-test='counter-title']`).text()
      expect(title).toBe('Event & Snapshot Test')
    })

    test('render the initial value 0', () => {
       let counter = wrapper.find(`[data-test='counter-value']`).text()
       expect(counter).toBe('0')
    })

    test('render the click event of increment button', () => {
       wrapper.find(`[data-test='increment-btn']`).simulate('click') 
       expect(wrapper.find(`[data-test='counter-value']`).text()).toBe('1')
    })

    test('render the click event of decrement button', () => {
       wrapper.find(`[data-test='decrement-btn']`).simulate('click')
       expect(wrapper.find(`[data-test='counter-value']`).text()).toBe('0')
    })

})


/**
 * toBe():-  It uses === to check strict equality.
 * 
 * toEqual():- It check that two objects have the same value
 */