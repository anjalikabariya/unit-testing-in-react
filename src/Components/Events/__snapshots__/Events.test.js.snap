// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Component Matches the snapshot 1`] = `
<div
  className="Counter"
>
  <h1
    data-test="counter-title"
  >
    Event & Snapshot Test
  </h1>
  <h2
    data-test="counter-value"
  >
    0
  </h2>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-test="increment-btn"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Increment
  </button>
  Â 
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-test="decrement-btn"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Decrement
  </button>
</div>
`;
